pkgs/by-name/si/simp/package.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  libxkbcommon,
+  vulkan-loader,
+  stdenv,
+  darwin,
+  wayland,
+  cargo-about,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "simp";
+  version = "3.5.3";
+
+  src = fetchFromGitHub {
+    owner = "Kl4rry";
+    repo = "simp";
+    rev = "v${version}";
+    hash = "sha256-NALVHcU+NhCT5w+pn6WhP+cFGPuDhnPt+IDNHN/oqls=";
+  };
+
+  cargoHash = "sha256-Wp50ldXGy7MLQTnvZDx3LIr8jLBpkoCrh0/xPowIn54=";
+
+  nativeBuildInputs = [
+    pkg-config
+    cargo-about
+  ];
+
+  buildInputs =
+    [
+      libxkbcommon
+      vulkan-loader
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.AppKit
+      darwin.apple_sdk.frameworks.CoreGraphics
+      darwin.apple_sdk.frameworks.CoreServices
+      darwin.apple_sdk.frameworks.Foundation
+      darwin.apple_sdk.frameworks.Metal
+      darwin.apple_sdk.frameworks.QuartzCore
+    ]
+    ++ lib.optionals stdenv.isLinux [ wayland ];
+
+  preBuild = ''
+    # replace variable from git cmd output
+    sed -i '35,39c let git_hash=String::from("${version}");' build.rs
+  '';
+
+  meta = with lib; {
```

```suggestion
  meta = {
```

tracking issue: #292468
discouraged as it leads to ambiguous `with`s

---

please read the example mentioned in the issue.

---

I did read the mentioned issue, but it seems that there is no consensus on the question if `with lib;` is good or bad.

---

https://github.com/NixOS/nixpkgs/issues/292468#issuecomment-2016454242

---

BTW there is no such thing as _unanimity_ in nixpkgs, but when 8 people including the person who created the issue (and originally believed that `meta = with lib;` was fine) thumbed it up, especially in comparison to the people who disagreed on _non-technical grounds_ and only claimed that keeping it "looks cleaner", I'd consider that overall _consensus_.

---

Thanks for the tip. This is is actually the first solid technical argument against `with lib;`.

> BTW there is no such thing as unanimity in nixpkgs, but when 8 people

Consensus does note require unanimity, but we had issues that had more agreement than 8 thumbs up.

---

I'm not really interested in arguing this on an individual PR, if you have a point to raise, please raise it on the issue itself